From aa0422d5a70a437136338f7155f25c9dafae3f5d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C5=82=20Dzi=C4=99giel?= <rafostar.github@gmail.com>
Date: Wed, 20 Jan 2021 11:33:39 +0100
Subject: [PATCH 1/2] matroska: Treat non-image structure as attachment

Otherwise each structure is named as GstTagImageInfo even if
it does not contain any images which is misleading.
---
 gst/matroska/matroska-read-common.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/gst/matroska/matroska-read-common.c b/gst/matroska/matroska-read-common.c
index 90d6e38e1..b553fd21f 100644
--- a/gst/matroska/matroska-read-common.c
+++ b/gst/matroska/matroska-read-common.c
@@ -851,9 +851,11 @@ gst_matroska_read_common_parse_attached_file (GstMatroskaReadCommon * common,
     }
 
     /* Set filename and description in the info */
-    if (info == NULL)
-      info = gst_structure_new_empty ("GstTagImageInfo");
-
+    if (info == NULL) {
+      const gchar *structure_name = (image_type != GST_TAG_IMAGE_TYPE_NONE) ?
+          "GstTagImageInfo" : "GstTagAttachmentInfo";
+      info = gst_structure_new_empty (structure_name);
+    }
     gst_structure_set (info, "filename", G_TYPE_STRING, filename, NULL);
     if (description)
       gst_structure_set (info, "description", G_TYPE_STRING, description, NULL);
-- 
GitLab


From a2246b2576cfb2afe0ae9878bc2d69969927aadf Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rafa=C5=82=20Dzi=C4=99giel?= <rafostar.github@gmail.com>
Date: Wed, 20 Jan 2021 11:53:51 +0100
Subject: [PATCH 2/2] matroska: Set image/attachment structure mimetype

Set structure mimetype to fix data detection by mimetype in other plugins.
---
 gst/matroska/matroska-read-common.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gst/matroska/matroska-read-common.c b/gst/matroska/matroska-read-common.c
index b553fd21f..628e19669 100644
--- a/gst/matroska/matroska-read-common.c
+++ b/gst/matroska/matroska-read-common.c
@@ -857,6 +857,7 @@ gst_matroska_read_common_parse_attached_file (GstMatroskaReadCommon * common,
       info = gst_structure_new_empty (structure_name);
     }
     gst_structure_set (info, "filename", G_TYPE_STRING, filename, NULL);
+    gst_structure_set (info, "mimetype", G_TYPE_STRING, mimetype, NULL);
     if (description)
       gst_structure_set (info, "description", G_TYPE_STRING, description, NULL);
 
-- 
GitLab

